src/PHPStan/Reflection/TypeToCallReflectionResolver/TypeToCallReflectionResolverRegistry.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPStan\Reflection\TypeToCallReflectionResolver;
+
+use function get_class;
+use PHPStan\Type\Type;
+
+final class TypeToCallReflectionResolverRegistry
+{
+    /**
+     * @var callable(class-string<Type>):?TypeToCallReflectionResolverInterface
+     */
+    private $factory;
+
+    /**
+     * @param callable(class-string<Type>):?TypeToCallReflectionResolverInterface $factory
+     */
+    public function __construct(callable $factory)
+    {
+        $this->factory = $factory;
+    }
+
+    public function getResolverFor(Type $type): ?TypeToCallReflectionResolverInterface
+    {
+        $factory = $this->factory;
+
+        return $factory(get_class($type));
```

Comment:
Callable objects are weak typed. Better use interface with methdo type, as in  https://github.com/rectorphp/rector/pull/2660